### PR TITLE
Improved UX around "Show solution" button in workshops & "Run" button in editor

### PR DESCRIPTION
**Pull request by Spyy004**
I have tried to improve the user experience by changing the color of text to white whenever the run and show solution button is disabled. 

There was an enhancement request for the show solution button. I saw the Run button also behaves similarly so I decided to add the same effect there as well. 

Would love to get feedback from you on my decision and code and if there aren't any issues then please merge the pull request.

### CODE REVIEW

**Comment by domesticmouse** [comment]
PTAL @johnpryan

**Comment by Spyy004** [comment]
Can someone please review this PR

**Comment by domesticmouse** [comment]
@johnpryan any idea when you might have time to review this? I'm guessing after I/O?

File: lib/elements/elements.dart [hunk 1]
```diff
@@ -111,6 +111,10 @@ class DButton extends DElement {
   set disabled(bool value) {
     buttonElement.disabled = value;
   }
+
+  set textColor(String color) {
+    buttonElement.style.color = 'white';
```
Comment (johnpryan) [inline lib/elements/elements.dart:116]: Should this be `= color`?

Comment (Spyy004) [reply lib/elements/elements.dart]: Ahh yes, I kept it white for testing and then forgot to make it as color. My bad. I have made the required changes. Please have a look

File: lib/workshops.dart [hunk 2]
```diff
@@ -180,6 +180,7 @@ class WorkshopUi extends EditorUi {
       if (solutionShownThisStep) {
         // Enable the show solution button
         showSolutionButton.disabled = false;
+        showSolutionButton.textColor = 'blue';
```
Comment (johnpryan) [inline lib/workshops.dart:183]: This doesn't match the original color:
Screenshot 2023-05-10 at 1 45 40 PM

Screenshot 2023-05-10 at 1 45 33 PM

**Comment by Spyy004** [comment]
What do you mean by Saas? I definitely want more info as I want to contribute to DartPad. Please tell me more about it

**Comment by domesticmouse** [comment]
> What do you mean by Saas? I definitely want more info as I want to contribute to DartPad. Please tell me more about it

Here ya go: https://sass-lang.com/

**Comment by Spyy004** [comment]
So do I use the Saas approach in changing colors and fix the above error that I made, and raise a new PR ?

**Comment by johnpryan** [comment]
That's a good question, the best way to contribute is to find an issue (or create one) and we can discuss the best way to fix it. I imagine if there are UX issues with the Workshop button, this would apply to the other buttons too.

**Pull request closed without merging**